lib/encrypt_mails.rb

```diff
@@ -151,51 +150,12 @@ def reset(header)
 
     end
 
-    def find_key_by_email(email)
-
-      user = User.find_by_mail(email)
-      user_key = Pgpkey.find_by user_id: user.id
-      unless user_key.nil?
-        return user_key.fpr
-      end
-
-    end
-
-    # handles headers for password reset emails, which have a single recipient
-    def password_reset_recipient(headers)
-
-      # hash to be returned
-      recipient = {
-        :encrypted => {:to => []},
-        :blocked => {:to => []},
-        :filtered  => {:to => []},
-        :unchanged => {:to => []},
-        :lost => {:to => []}
-      }
-
+    def password_reset_headers(headers)
       user = User.find_by_mail(headers[:to])
-      user_key = Pgpkey.find_by(user_id: user.id)
-
-      if user_key.nil?
-        logger.info "No public key found for #{user} <#{user.mail}> (#{user.id})" if logger
-      else
-        recipient[:encrypted][:to].push user
-      end
-
-      # unencrypted
-      case Setting.plugin_openpgp['unencrypted_mails']
-        when 'blocked'
-          recipient[:blocked][:to].push user
-        when 'filtered'
-          recipient[:filtered][:to].push user if user_key.nil?
-        when 'unchanged'
-          recipient[:unchanged][:to].push user if user_key.nil?
-        else
-          recipient[:lost][:to].push user
-      end
-
-      recipient
-
+      users = Array.new
+      users.push user
+      headers[:to] = users
```

Comment:
This whole function could be one line: `headers[:to] = [User.find_by_mail(headers[:to])]`.

There should also be a comment explaining why this is being done.